sale_force_invoiced/model/sale_order.py

```diff
@@ -21,6 +20,6 @@ class SaleOrder(models.Model):
     def _compute_invoice_status(self):
         res = super()._compute_invoice_status()
         self.filtered(
-            lambda so: so.force_invoiced and so.state in ("sale", "done")
+            lambda so: so.force_invoiced and (so.state == "sale" or so.locked)
```

Comment:
In principle, only orders in `sale` status can be locked. I think you can simplify the module by ignoring the `locked` field everywhere and just go by state `sale`. Note that orders in any other state will not be marked as invoiceable by Odoo: https://github.com/odoo/odoo/blob/3f6deca/addons/sale/models/sale_order.py#L530-L531 so even if orders were to be locked in any other status, this override might be considered meaningless for such orders. Or am i missing anything?